config/webpack.prod.config.js

```diff
@@ -69,38 +69,20 @@ module.exports = Merge.smart(commonConfig, {
       },
     ],
   },
+  // New in Webpack 4. Replaces CommonChunksPlugin. Extract common modules among all chunks to one
+  // common chunk and extract the Webpack runtime to a single runtime chunk.
+  optimization: {
+    runtimeChunk: 'single',
+    splitChunks: {
+      chunks: 'all',
+    },
+  },
   // Specify additional processing or side-effects done on the Webpack output bundles as a whole.
   plugins: [
-    // CommonsChunkPlugin creates separate files with common modules shared between multiple entry
-    // points.
-    //
-    // This extracts all modules from node_modules to a common bundle called "vendor". It can be
-    // cached by users' browsers long-term and will only change if we upgrade a package.
-    new webpack.optimize.CommonsChunkPlugin({
-      name: 'vendor',
-      minChunks: module => module.context && module.context.includes('node_modules'),
-    }),
-    // Extracts the webpack bootstrap logic to a separate file. The contents of this file can change
-    // after every build, but are small in size.
-    new webpack.optimize.CommonsChunkPlugin({
-      name: 'manifest',
-      minChunks: Infinity,
-    }),
     // Writes the extracted CSS from each entry to a file in the output directory.
     new ExtractTextPlugin({
       filename: '[name].min.css',
       allChunks: true,
     }),
-    // Defines a global variable that all bundles can access.
-    new webpack.DefinePlugin({
-      'process.env': {
-        // Maps the env var NODE_ENV at compile time to a global var available to bundles at
-        // runtime.  Should be set to "production" so that JS code knows it is running in a
-        // production environment.
-        NODE_ENV: JSON.stringify(process.env.NODE_ENV),
-      },
-    }),
-    // Minifies the JS.
-    new webpack.optimize.UglifyJsPlugin({ sourceMap: true }),
```

Comment:
mode: 'production' also does this by default.

Reply:
I know somebody can just "read the docs" but do you mind adding a comment about things that are defaulted? One of the things that was hard to keep track of was all the default behaviors that Webpack has.

Reply:
The funny thing is that there currently no docs on Webpack 4 and what `mode: production` does by default besides [this medium post](https://medium.com/webpack/webpack-4-mode-and-optimization-5423a6bc597a) :( .

I can try to enumerate everything in a comment here.

Reply:
![alt-text](https://media.giphy.com/media/AaosjRHKjEcXm/giphy.gif)